test/e2e/scaling_dbsize_test.go

```diff
@@ -0,0 +1,82 @@
+package e2e
+
+import (
+	"context"
+	"github.com/openshift/cluster-etcd-operator/pkg/etcdenvvar"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/cluster-etcd-operator/test/e2e/framework"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	operatorNamespace      = "openshift-etcd-operator"
+	operatorDeploymentName = "etcd-operator"
+	operatorUpdatedImage   = "quay.io/melbeher/ceo-1900"
+	operatorImageKey       = "OPERATOR_IMAGE"
+	operandNameSpace       = "openshift-etcd"
+	etcdCRName             = "cluster"
+	etcdLabelSelector      = "app=etcd"
+	quotaEnvName           = "ETCD_QUOTA_BACKEND_BYTES"
+	etcdPodName            = "etcd"
+)
+
+func TestEtcdDBScaling(t *testing.T) {
+	dbSize32GB := int32(32)
+	opClientSet := framework.NewOperatorClient(t)
+	clientSet := framework.NewClientSet("")
+
+	etcdCR, err := opClientSet.OperatorV1().Etcds().Get(context.Background(), etcdCRName, metav1.GetOptions{})
+	require.NoError(t, err)
+
+	etcdCR.Spec.BackendQuotaGiB = dbSize32GB
+	etcdCR, err = opClientSet.OperatorV1().Etcds().Update(context.Background(), etcdCR, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	var etcdPod v1.Pod
+	// wait for new etcd pods to be rolled out
+	err = wait.PollUntilContextTimeout(context.Background(), time.Second, 10*time.Minute, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
there's a nicer helper in library-go that's waiting for revision rollouts:

https://github.com/openshift/library-go/blob/52527b886e57ab47fb355a4d1234200047996004/test/library/pod_same_revision.go#L24

I believe we might be using it here for something, we definitely use it a lot in origin

Reply:
yes we used it in `vertical-scaling` test 

in this current context, we just check the first etcd-pod that is `running`, we dont need all the revision to be rolled-out imho

but I change it no issue 👍🏽

Reply:
any code that we don't maintain is good code ;)

Reply:
lmfaoooo :D :D :D :D :D :D :D